### PR TITLE
Test Refactor

### PR DESCRIPTION
**Pull request by carlospaelinck**
Addresses #288 

- [x] Refactors most tests to a Jest/Enzyme environment
- [x] Adds a `start` task for development mode
- [x] Adds testing back to CI

cc @kenwheeler @ryan-roemer

### CODE REVIEW

**Comment by ryan-roemer** [comment]
@carlospaelinck -- Do we need `gulp` as a dep any more?

File: package.json [hunk 1]
```diff
@@ -68,5 +77,11 @@
     "url": "https://github.com/kenwheeler/nuka-carousel/issues"
   },
   "homepage": "https://github.com/kenwheeler/nuka-carousel",
-  "sideEffects": false
+  "sideEffects": false,
+  "jest": {
+    "setupFiles": [
+      "<rootDir>/test/helpers/raf-polyfill.js",
+      "<rootDir>/test/helpers/jest-setup.js"
+    ]
```
Comment (blainekasten) [inline package.json]: If you want to simplify setup you can use enzyme-matchers. new 5.0.0 version has a `jest-environment-enzyme` which greatly simplifies setup. 

* removes the requirement to install enzyme/adapter
* removes the requirement for your own enzyme adapter setup and raf polyfill

Comment (carlospaelinck) [reply package.json]: Oh nice. I'll update this PR with that!

File: package.json [hunk 2]
```diff
@@ -68,5 +79,9 @@
     "url": "https://github.com/kenwheeler/nuka-carousel/issues"
```
Comment (ryan-roemer) [inline package.json]: Unrelated, but can you update the `https://github.com/kenwheeler/` strings to `https://github.com/FormidableLabs/`? (missed from when Ken ported over the repo).

**Comment by carlospaelinck** [comment]
@ryan-roemer Updated the strings

**Pull request merged**